### PR TITLE
Retrying and limiting messages length when sending to teams

### PR DESCRIPTION
**Pull request by blalop**
### Description of the Change

In the current state of prom2teams, there is nothing it can do to prevent missing Prometheus alerts due to several reasons, such as MS Teams being not available or prom2teams trying to send a payload that exceeds the limit MS Teams has.

### Benefits

No alerts will be lost due to:
* Teams being down
* Sending a payload that is too long

### Possible Drawbacks

None

### Applicable Issues

#223

### CODE REVIEW

**Review by jmonterrubio — APPROVED** [review]

File: prom2teams/app/teams_client.py [hunk 1]
```diff
@@ -1,20 +1,55 @@
...
-        exception_msg = 'Error performing request to: {}.\n' \
-                        ' Returned status code: {}.\n' \
-                        ' Returned data: {}\n' \
-                        ' Sent message: {}\n'
-        raise MicrosoftTeamsRequestException(exception_msg.format(teams_webhook_url,
-                                                                  str(response.status_code),
-                                                                  str(response.text),
-                                                                  str(message)),
-                                             code=response.status_code)
+logger = logging.getLogger('prom2teams')
```
Comment (vicsufer) [inline prom2teams/app/teams_client.py:8]: Other files call the logger variable "log", it is better if we maintain the same naming convention.

**Comment by vicsufer** [comment]
Question time, how is this interacting when using multiple connectors? 
- The same configuration is applied for all of them? (I think so)
- If a request fails into one connector, the retry is only applied to this connector or will it trigger an unnecesary retry to the others?

**Comment by blalop** [comment]
Hi @vicsufer:

* You're right, this PR does not provide per-connector configuration.
* As the retry is managed at the teams client level - it has no knowledge of connectors. This means that it will only retry the failed connections.

**Review by vicsufer — APPROVED** [review]

**Review by dortegau — APPROVED** [review]

**Review by pablogcaldito — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -214,6 +214,11 @@ Excluded: <Coma separated list of labels to ignore>
 
 [Annotations]
 Excluded: <Comma separated list of annotations to ignore>
+
```
Comment (dortegau) [inline README.md:217]: It would be nice to describe this key feature before :)

**Pull request merged**